frappe/api/__init__.py

```diff
@@ -0,0 +1,64 @@
+# Copyright (c) 2015, Frappe Technologies Pvt. Ltd. and Contributors
+# License: MIT. See LICENSE
+from werkzeug.exceptions import NotFound
+from werkzeug.routing import Map, Submount
+from werkzeug.wrappers import Request, Response
+
+import frappe
+import frappe.client
+from frappe import _
+from frappe.utils.response import build_response
+
+
+def handle(request: Request):
+	"""
+	Entry point for `/api` methods.
+
+	APIs are versioned using second part of path.
+	v1 -> `/api/v1/*`
+	v2 -> `/api/v2/*`
+
+	Different versions have different specification but broadly following things are supported:
+
+	- `/api/method/{methodname}` will call a whitelisted method
+	- `/api/resource/{doctype}` will query a table
+	        examples:
+	        - `?fields=["name", "owner"]`
+	        - `?filters=[["Task", "name", "like", "%005"]]`
+	        - `?limit_start=0`
+	        - `?limit_page_length=20`
+	- `/api/resource/{doctype}/{name}` will point to a resource
+	        `GET` will return document
+	        `POST` will insert
+	        `PUT` will update
```

Comment:
`PATCH` could be implemented to update a few fields. Contrary to `PUT`, you can get away by passing a sparse Document?

Reply:
OK, currently PUT == PATCH (update_doc implementation)

Reply:
Frappe's `PUT` IS `PATCH` if we go by conventions.

Reply:
https://github.com/frappe/frappe/pull/22300/commits/1834488f6174ad4f32df164822c85b2c20dfeb34